crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -380,20 +388,11 @@ impl Account {
         match self.parse_decrypted_to_device_event(sender, sender_key, plaintext).await {
             Ok(result) => Ok((session, result)),
             Err(e) => {
-                // We might created a new session but decryption might still
+                // We might have created a new session but decryption might still
                 // have failed, store it for the error case here, this is fine
                 // since we don't expect this to happen often or at all.
                 match session {
-                    SessionType::New(s) => {
-                        let account = self.store.account();
-                        let changes = Changes {
-                            account: Some(account.clone()),
-                            sessions: vec![s],
-                            ..Default::default()
-                        };
-                        self.store.save_changes(changes).await?;
-                    }
-                    SessionType::Existing(s) => {
+                    SessionType::New(s) | SessionType::Existing(s) => {
```

Comment:
Hmm, a little bit uneasy about this one. We're now persisting the `Session` but not the `Account`. This means that the `Account` will contain the one-time key to create the session for a bit longer than needed.

I guess there's nothing we can do here until we move the `Session` to use the transaction as well.